internal/cmd/exodus.go

```diff
@@ -136,6 +136,29 @@ func exodusMain(ctx context.Context, cfg conf.Config, args args.Config) int {
 		return 73
 	}
 
+	var publish gw.Publish
+
+	logger.F("items", len(items)).Info("Preparing to publish items")
+
+	if args.Publish == "" {
+		// No publish provided, then create a new one.
+		publish, err = gwClient.NewPublish(ctx)
+		if err != nil {
+			logger.F("error", err).Error("can't create publish")
+			return 62
+		}
+		logger.F("publish", publish.ID()).Info("Created publish")
+	} else {
+		publish = gwClient.GetPublish(args.Publish)
+		err = publish.Get(ctx)
+		// No existing publish object from database, then will not do uploading.
+		if err != nil {
+			logger.F("error", err).Error("can't join publish")
+			return 62
```

Comment:
For this errno `62 (ETIME, Timer expired)`, I think it's similar to the above "can't create publish" error, so I chose 62 for "can't join publish" error as well

Reply:
I appreciate you sharing your reasoning and wanting to be consistent.

I think it'd be better to use unique exit codes though.  The point of using unique exit codes for failure, rather than always using just `1`, is that sometimes you might find yourself in a situation where you have only the exit code and no other information about a failure.

Imagine for example that exodus-rsync was run by some other code which did not capture stdout/stderr and didn't have a valid logging config for exodus-rsync.  You might get back an error message along the lines of `Command 'exodus-rsync ...' exited with code 62` and no other details.  Ideally then, you can just "git grep 62" and figure out exactly which error case happened.  If the same exit code is used for two error cases then you'll find two possible cases and you can't be certain which one of them occurred.  So it'd be better to use a unique value at each error path.

Also, there is no direct relationship between the exit codes used and the possible values of `errno`.